src/components/features/EditPostForm/EditPostForm.js

```diff
@@ -0,0 +1,35 @@
+import { useDispatch } from 'react-redux';
+import { editPost } from '../../../redux/postsRedux';
+import { useNavigate } from "react-router-dom";
+import PostForm from '../PostForm/PostForm';
+import { useSelector } from 'react-redux';
+import { getPostById } from '../../../redux/postsRedux';
+
+
+const EditPostForm = ({id}) => {
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  
+  const postData = useSelector(state => getPostById(state, id));
+ 
+
+
+  const handleSubmit = (props) => {
+    dispatch(editPost({ ...props, id }));
+    navigate("/", { replace: true });
+  }
+
+  return (
+    <PostForm
```

Comment:
Nie ma potrzeby, aby wyciągać cały jeszcze ten formularz do zewnętrznego komponentu. To co przekazujesz do zmiennych `initial*`, po prostu ustaw to jako wartość początkową dla useState